app/src/main/java/pm/gnosis/heimdall/ui/safe/details/SafeDetailsFragment.kt

```diff
@@ -62,8 +61,11 @@ class SafeDetailsFragment : BaseFragment() {
         safeAddress = arguments?.getString(EXTRA_SAFE_ADDRESS)?.asEthereumAddress()!!
         viewModel.setup(safeAddress)
 
-        layout_safe_details_send_button.setCompoundDrawableResource(left = R.drawable.ic_send_azure)
-        layout_safe_details_receive_button.setCompoundDrawableResource(left = R.drawable.ic_qrcode_scan_azure)
+        layout_safe_details_safe_image.setAddress(safeAddress)
+
+        val addressString = safeAddress.asEthereumAddressChecksumString()
+        layout_safe_details_safe_address.text = "${addressString.subSequence(0, 4)}...${addressString.subSequence(addressString.length - 4, addressString.length)}"
```

Comment:
Should we create an extension for this? It will probably be used more often

Reply:
Extension function added